### PR TITLE
only call onPanelMigration when the version actually changes

### PR DESCRIPTION
**Pull request by ryantxu**
The `onPanelMigration` handler should only be called if the version changes.  

We should set the version based on the current grafana build number if one is not specified in the plugin info.  None of the builtin panels set a plugin version so it should be the system version, not 1.0.0

Alternativly, `PanelPlugin` could expose an `optionsVersion` that only changes when the options model changes.  many updates to the plugin keep the options the same and do not need a callback.  That may be more complex than necessary though :)

### CODE REVIEW

**Comment by ryantxu** [comment]
FYI, the existing call:
```
this.pluginVersion = plugin.info ? plugin.info.version : '1.0.0';
```

returns empty string '' for all built-in panels

**Comment by ryantxu** [comment]
@torkelo -- this patch fixes a few issues with your previous cleanup.  In particular, checking for null values of existing options (totally possible with new panels or migrating from angular)

It also fixed the logic about cleaning up angular panels.  It now checks if the *previous* panel was angular, not the new one.

There is still something funky with switch between angular panels.  I get messages like this often when showing singlestat:
image

File: public/app/features/dashboard/state/PanelModel.ts [hunk 1]
```diff
@@ -262,7 +271,7 @@ export class PanelModel {
     const reactPanel = newPlugin.exports.reactPanel;
 
     // for angular panels we must remove all events and let angular panels do some cleanup
-    if (!reactPanel) {
+    if (!_.has(this, 'plugin.exports.reactPanel')) {
```
Comment (torkelo) [inline public/app/features/dashboard/state/PanelModel.ts]: ah, good fix! but really not loving these lodash get/has, now that we have such fine typing it's a shame to use things like this.

Comment (ryantxu) [reply public/app/features/dashboard/state/PanelModel.ts]: is:
```
if (!(this.plugin && this.plugin.exports &&  this.plugin.exports && this.plugin.exports.reactPanel) 
```
better?  😱 

Is there a good typed way to check these chains of possible null/missing values?

**Pull request merged**